### DISCUSSION TITLE
Execute only one rule based on salience

### DISCUSSION THREAD

**Discussion by Madhukarc** [hyperjumptech/grule-rule-engine#414] [Q&A]
Hi,

I have a scenario where for a given fact system finds more than one matching rules. I want to execute only one rule which has highest priority and dont want to execute remaining. 
If I set the maxCycle as 1, it does the desired thing, but I am not sure this is the right way to do or not. 
Is there any recommended way to achieve the same.

Thanks in Advance.
Madhukar

**Comment by djg2002** [comment]
I've been using Retract() for other rules I don't want

**Comment by lucklyt** [comment]
you can use Complete()  in then  for every rule
